neo/SmartContract/Manifest/ContractManifest.cs

```diff
@@ -0,0 +1,193 @@
+        /// </summary>
+        public ContractFeatures Features { get; set; }
+
+        /// <summary>
+        /// For technical details of ABI, please refer to NEP-3: NeoContract ABI. (https://github.com/neo-project/proposals/blob/master/nep-3.mediawiki)
+        /// </summary>
+        public ContractAbi Abi { get; set; }
+
+        /// <summary>
+        /// The permissions field is an array containing a set of Permission objects. It describes which contracts may be invoked and which methods are called.
+        /// </summary>
+        public WildCardContainer<ContractPermission> Permissions { get; set; }
+
+        /// <summary>
+        /// The trusts field is an array containing a set of contract hashes or group public keys. It can also be assigned with a wildcard *. If it is a wildcard *, then it means that it trusts any contract.
+        /// If a contract is trusted, the user interface will not give any warnings when called by the contract.
+        /// </summary>
+        public WildCardContainer<UInt160> Trusts { get; set; }
+
+        /// <summary>
+        /// The safemethods field is an array containing a set of method names. It can also be assigned with a wildcard *. If it is a wildcard *, then it means that all methods of the contract are safe.
+        /// If a method is marked as safe, the user interface will not give any warnings when it is called by any other contract.
+        /// </summary>
+        public WildCardContainer<string> SafeMethods { get; set; }
+
+        /// <summary>
+        /// Create Default Contract manifest
+        /// </summary>
+        /// <param name="hash">Hash</param>
+        /// <returns>Return default manifest for this contract</returns>
+        public static ContractManifest CreateDefault(UInt160 hash)
+        {
+            return new ContractManifest()
+            {
+                Permissions = WildCardContainer<ContractPermission>.CreateWildcard(),
+                Abi = new ContractAbi()
+                {
+                    Hash = hash,
+                    EntryPoint = ContractMethodDescriptor.DefaultEntryPoint,
+                    Events = new ContractEventDescriptor[0],
+                    Methods = new ContractMethodDescriptor[0]
+                },
+                Features = ContractFeatures.NoProperty,
+                Groups = new ContractGroup[0],
+                SafeMethods = WildCardContainer<string>.Create(),
+                Trusts = WildCardContainer<UInt160>.Create()
+            };
+        }
+
+        /// <summary>
+        /// Return true if is allowed
+        /// </summary>
+        /// <param name="manifest">Manifest</param>
+        /// <param name="method">Method</param>
+        /// <returns>Return true or false</returns>
+        public bool CanCall(ContractManifest manifest, string method)
+        {
+            if (Groups != null && manifest.Groups != null && Groups.Any(a => manifest.Groups.Any(b => a.PubKey.Equals(b.PubKey))))
```

Comment:
@erikzhang  `Groups` never will be null now?

Reply:
Emtpy is allowed, but `null` is not allowed.

Reply:
My original proposal was `group` instead of `groups`. In that case, `null` is allowed. For `groups` , we should use empty array.